### PR TITLE
chore: update webdriverio

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

Updates WebDriverIO. Due to [some changes](https://webdriver.io/blog/2023/07/31/driver-management/) to how it handles browser drivers, we needed to change our config to make it work.

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

File: packages/@lwc/integration-tests/scripts/build.js [hunk 1]
```diff
@@ -112,7 +111,10 @@ testEntries
             templates.html(testName),
             'utf8'
         );
-    }, Promise.resolve())
-    .catch((err) => {
-        console.log(err);
-    });
+    }
+}
```
Comment (nolanlawson) [inline packages/@lwc/integration-tests/scripts/build.js:115]: I noticed some errors in CircleCI; this should fix it by 1) not using `fs-extras`, which isn't even a direct dependency of ours, and 2) doing all the builds/writes serially.

```
(node:2027) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 19 error listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit

```

File: .circleci/config.yml
Comment (nolanlawson) [inline .circleci/config.yml:1]: I couldn't for the life of me figure out what was wrong with the local Chrome on Circle CI. So I just moved us back to using SauceLabs instead.

The downside is we can't do memory leak testing, but that's been busted anyway: #3393

File: .circleci/config.yml [hunk 2]
```diff
@@ -319,25 +319,28 @@ jobs:
...
     steps:
-      - browser-tools/install-browser-tools # install local chrome rather than using saucelabs
       - load_workspace
+      - start_sauce_connect
       - retry_command:
-          command_name: Run integration test - local Chrome - dev mode
-          command: yarn local:dev
+          command_name: Run integration test - Chrome - dev mode
+          command: yarn sauce:dev --browsers chrome
           working_directory: packages/@lwc/integration-tests
```
Comment (nolanlawson) [inline .circleci/config.yml:344]: I couldn't for the life of me figure out what was wrong with the local Chrome on Circle CI. So I just moved us back to using SauceLabs instead.

The downside is we can't do memory leak testing, but that's been busted anyway: https://github.com/salesforce/lwc/issues/3393

File: packages/@lwc/integration-tests/scripts/wdio.local.conf.js [hunk 3]
```diff
@@ -23,23 +18,13 @@ exports.config = merge(baseConfig.config, {
         {
             browserName: 'chrome',
             'goog:chromeOptions': {
-                args: headless ? ['headless', 'disable-gpu'] : [],
+                args: headless ? ['headless=new'] : [],
```
Comment (nolanlawson) [inline packages/@lwc/integration-tests/scripts/wdio.local.conf.js:21]: May as well switch to the new one. This is only for local testing though.

File: packages/@lwc/integration-tests/scripts/wdio.local.conf.js [hunk 4]
```diff
@@ -23,23 +18,13 @@ exports.config = merge(baseConfig.config, {
...
-            'selenium-standalone',
-            {
-                installArgs: {
-                    drivers,
-                },
-                args: {
-                    drivers,
-                },
-            },
-        ],
```
Comment (nolanlawson) [inline packages/@lwc/integration-tests/scripts/wdio.local.conf.js:42]: Having any `selenium-standalone` config here borked us with the wdio upgrade.

**Review by ekashida — APPROVED** [review]

**Pull request merged**